core/src/lib/address_conf.cc

```diff
@@ -341,6 +339,58 @@ bool SetupPort(unsigned short& port,
   }
 }
 
+bool IsFamilyEnabled(IpFamily fam)
+{
+  static bool family_enabled[] = {CheckIfFamilyEnabled(IpFamily::V4),
+                                  CheckIfFamilyEnabled(IpFamily::V6)};
+  return family_enabled[static_cast<std::underlying_type_t<IpFamily>>(fam)];
+}
+
+std::optional<int> GetFamily(IpFamily family)
+{
+  switch (family) {
+    case IpFamily::V4:
+      return AF_INET;
+    case IpFamily::V6:
+      return AF_INET6;
+    default:
+      return std::nullopt;
+  }
+}
+
+std::string FamilyName(IpFamily fam)
+{
+  switch (fam) {
+    case IpFamily::V4:
+      return "IPv4";
+    case IpFamily::V6:
+      return "IPv6";
+    default:
+      return "*Unknown Protocol*";
+  }
```

Comment:
Should we throw an exception here instead of returning normally ?  If we change the function to constexpr, then the compiler will catch those mistakes!
See https://godbolt.org/z/qf5hadP9Y for an example